File: 11.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    Solution() {
        ios::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);
    }
    int maxArea(vector<int>& height) {
        int res = -1, left = 0, right = height.size() - 1;
        while (left <= right) {
            res = max(res, (right - left) * min(height[left], height[right]));
            if (height[left] < height[right]) {
                ++left;
            } else {
                --right;
            }
        }
        return res;
    }
};

File: 11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        res = abs(height[left] - height[right])
        res = 0
        while right > left:
            res = max(res, (right - left) * min(height[right], height[left]))
            if height[right] > height[left]:
                left += 1
            else:
                right -= 1
        return res
